### PR TITLE
Add `no-array-push-push` rule

### PR DESCRIPTION
**Pull request by fisker**
Part of #937, only work for `.push()`

### CODE REVIEW

File: rules/no-array-push-push.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			const index = value.indexOf(node);
+
+			if (index !== -1) {
+				return value[index - 1];
+			}
+		}
+	}
+
+	/* istanbul ignore next */
+	throw new Error('Can not find the first `Array#push()` call.');
```
Comment (sindresorhus) [inline rules/no-array-push-push.js:49]:
```suggestion
	throw new Error('Cannot find the first `Array#push()` call.');
```

Comment (sindresorhus) [reply rules/no-array-push-push.js:49]: Should it really throw? If it is supposed to catch a case that should not happen, I think we should have some text that tells the user to open an issue.

File: docs/rules/no-array-push-push.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Enforce combine multiple `Array#push()` into one call
```
Comment (sindresorhus) [inline docs/rules/no-array-push-push.md:1]:
```suggestion
# Enforce combining multiple `Array#push()` into one call
```

File: docs/rules/no-array-push-push.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+[`Array#push()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) accepts multiple arguments, multiple calls should combine into one.
```
Comment (sindresorhus) [inline docs/rules/no-array-push-push.md:3]:
```suggestion
[`Array#push()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) accepts multiple arguments. Multiple calls should be combined into one.
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Not sure about the rule name. How about `no-multiple-array-push`?

**Comment by fisker** [comment]
I like `no-array-push-push`, like core rule [`no-plusplus`](https://eslint.org/docs/rules/no-plusplus).

File: rules/no-array-push-push.js (hunk 1 above)
Comment (fisker) [reply rules/no-array-push-push.js:49]: This function is a little bit hacky, I'm not sure if it can cover all cases, I can add a issue link.

**Comment by sindresorhus** [comment]
Ok. Sure. I don't feel strongly about it.

**Comment by fisker** [comment]
#1019

**Pull request merged**